kb/2023-02-01/scan_and_repair_filesystem.md

```diff
@@ -0,0 +1,125 @@
+---
+title: Scan and Repair Root Filesystem of VirtualMachine
+description: Scan and repair root filesystem of VM
+slug: scan-and-repair-vm-root-filesystem
+authors:
+  - name: Vicente Cheng
+    title: Senior Software Engineer
+    url: https://github.com/Vicente-Cheng
+    image_url: https://github.com/Vicente-Cheng.png
+tags: [storage, longhorn, root, filesystem]
+hide_table_of_contents: false
+---
+
+In earlier versions of Harvester (v1.0.3 and prior), Longhorn volumes sometimes get corrupted when rebuilding (reference: [Analysis: Potential Data/Filesystem Corruption](https://longhorn.io/kb/troubleshooting-volume-filesystem-corruption/#solution)). In Harvester v1.1.0 and later versions, the Longhorn team has fixed this problem. So, you can follow the steps below to scan the VM's filesystem and repair it if needed.
+
+
+## Stop The VM And Backup Volume
+
+Before we scan the filesystem, we recommend you back up the volume first. For example, refer to the following steps to stop the VM and backup the volume.
+
+- Find the target VM
+
+![finding the target VM](./imgs/finding_the_target_vm.png)
+
+- Stop the target VM
+
+![Stop the target VM](./imgs/stop_the_target_vm.png)
+
+Now, the VM stopped, and related volumes are detached. Then we go to the longhorn UI to backup this volume.
+
+- Enable `Developer Tools & Features` (Preferences -> Enable Developer Tools & Features)
+
+![Preferences then enable developer mode](./imgs/preferences_enable_developer_mode.png)
+![Enable the developer mode](./imgs/enable_the_developer_mode.png)
+
+- Go to edit config page of VM
+
+![goto edit config page of VM](./imgs/goto_vm_edit_config_page.png)
+
+- Choose volume and Click `Check volume details`
+
+![link to longhorn volume page](./imgs/link_to_longhorn_volume.png)
+
+- Attach this volume again.
```

Comment:
Go to `?` and select `Attach` to attach the volume again.

Reply:
Hmm, only a dropdown menu icon here. How about
`Click the dropdown menu on the right side and attach this volume again`?

Reply:
@LucasSaintarbor, ditto. Thanks for giving a review!

Reply:
@Vicente-Cheng confirming if this is step good? Thanks

Reply:
hi @LucasSaintarbor.
Ah, I see it. We have already fixed it. Thanks!!